src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.ClientCertProvider.cs

```diff
@@ -0,0 +1,201 @@
+
+                _certificateFile = null;
+                switch (s_clientCertificatePolicy)
+                {
+                    case ClientCertificatePolicy.Callback:
+                        {
+                            SetCallbacks(easy);
+                            break;
+                        }
+                    case ClientCertificatePolicy.CommonName :
+                        {
+                            SetClientCertCommonName(easy);
+                            break;
+                        }
+                    default:
+                        ExportToFile(easy);
+                        break;
+                }
+            }
+
+            public void Dispose()
+            {
+                if (_certificateFile == null)
+                {
+                    return;
+                }
+                try
+                {
+                    File.Delete(_certificateFile);
+                }
+                catch (IOException e)
+                {
+                    VerboseTrace("Deleting temporary file:" + e);
+                }
+            }
+
+            internal static ClientCertProvider Create(ClientCertificateOption option, string uriScheme)
+            {
+                if (option == ClientCertificateOption.Manual || !string.Equals(uriScheme, UriSchemeHttps))
+                {
+                    return null;
+                }
+
+                using (var myStore = new X509Store())
+                {
+                    myStore.Open(OpenFlags.ReadOnly);
+                    X509Certificate2Collection clientCerts = myStore.Certificates.FindTlsClientCertificates();
+                    if (clientCerts.Count == 0)
+                    {
+                        VerboseTrace("0 Client Certs");
+                        return null; 
+                    }
+                    VerboseTrace("Client Certs found " + clientCerts.Count);
+
+                    // NOTE: This is a DEVIATION FROM WinHttpHandler
+                    // Unlike WinHttpHandler, we are going to use only one client certificate.
+                    // libcurl, libssl and even X509Certificate2Collection.Export all are able to handle single client certificate.
+                    // Picking the first available cert
+                    return new ClientCertProvider(clientCerts[0]);
+                }
+            }
+
+            private static int CurlSslCallback(IntPtr easyHandle, IntPtr sslContext, IntPtr x509Handle)
+            {
+                Debug.Assert(x509Handle != null);
+                Debug.Assert(sslContext != null);
+
+                Interop.libcurl.SSL_CTX_use_certificate(sslContext, x509Handle);
+
+                // TODO: Need to set the private key. Fix when dev_sslstream gets merged.
+                // For now setting it to nullPtr
+                Interop.libcurl.SSL_CTX_use_PrivateKey(sslContext, IntPtr.Zero);
```

Comment:
I was thinking we can use [Set.SslCertificate](https://github.com/dotnet/corefx/blob/dev-sslstream/src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs#L345). In that case, I will pass cert and pointer in one single call.